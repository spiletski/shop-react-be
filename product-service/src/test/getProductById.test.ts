import {getProductById} from "@functions/getProductById/handler";

const event = {
    pathParameters: {
        productId: "aaa",
    },
};

describe('getProductById handler', function () {
    it('verifies successful response', async () => {

        // @ts-ignore
        const result = await getProductById(event)

        expect(result.statusCode).toEqual(200);
        expect(result.body).toMatchSnapshot();
    });

    it("product not found", async () => {
        // @ts-ignore
        const actualValue = await getProductById({
            pathParameters: {
                productId: "aaa-123",
            }
        });
        expect(actualValue).toMatchSnapshot();
    });
});