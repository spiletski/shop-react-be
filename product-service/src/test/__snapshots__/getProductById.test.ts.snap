// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProductById handler product not found 1`] = `
Object {
  "body": "{\\"error\\":\\"Auto with aaa-123 not found\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Methods": "OPTIONS,GET",
    "Access-Control-Allow-Origin": "*",
    "content-type": "application/json",
  },
  "statusCode": 404,
}
`;

exports[`getProductById handler verifies successful response 1`] = `"{\\"auto\\":{\\"id\\":\\"aaa\\",\\"Name\\":\\"chevrolet camaro\\",\\"Miles_per_Gallon\\":27,\\"Cylinders\\":4,\\"Displacement\\":151,\\"Horsepower\\":90,\\"Weight_in_lbs\\":2950,\\"Acceleration\\":17.3,\\"Year\\":\\"1982-01-01\\",\\"Origin\\":\\"USA\\"}}"`;

exports[`getProductByIdt handler if error should return response 500 response 1`] = `
Object {
  "body": "{\\"error\\":\\"Auto with aaa-123 not found\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Methods": "OPTIONS,GET",
    "Access-Control-Allow-Origin": "*",
    "content-type": "application/json",
  },
  "statusCode": 404,
}
`;

exports[`getProductByIdt handler product not found 1`] = `
Object {
  "body": "{\\"error\\":\\"Auto with aaa-123 not found\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Methods": "OPTIONS,GET",
    "Access-Control-Allow-Origin": "*",
    "content-type": "application/json",
  },
  "statusCode": 404,
}
`;

exports[`getProductByIdt handler verifies successful response 1`] = `"{\\"auto\\":{\\"id\\":\\"aaa\\",\\"Name\\":\\"chevrolet camaro\\",\\"Miles_per_Gallon\\":27,\\"Cylinders\\":4,\\"Displacement\\":151,\\"Horsepower\\":90,\\"Weight_in_lbs\\":2950,\\"Acceleration\\":17.3,\\"Year\\":\\"1982-01-01\\",\\"Origin\\":\\"USA\\"}}"`;
